File: solutions/0779. K-th Symbol in Grammar/0779.cs
// Time complexity: O(n)
// Space complexity: O(n)
public class Solution {
    public int KthGrammar(int n, int k) {
        string binaryString = Convert.ToString(k - 1, 2);
        int count = binaryString.Replace("0", "").Length;
        return count % 2 == 0 ? 0 : 1;
    }
}

File: solutions/0779. K-th Symbol in Grammar/0779.js
// Time complexity: O(logk)
// Space complexity: O(logk)
var kthGrammar = function(n, k) {
    let count = (k - 1).toString(2).split('0').join('').length;
    return count % 2 === 0 ? 0 : 1;
};